packages/backend-common/src/reading/GithubUrlReader.ts

```diff
@@ -342,18 +312,34 @@ export class GithubUrlReader implements UrlReader {
     return repo.default_branch;
   }
 
-  private async fetchResponse(
+  protected async fetchResponse(
```

Comment:
Why was this changed? It changes the public interface, which we probably don't want

Reply:
To allow overwriting this via an class extension. 

Currently we create our own GithubUrlReader which overwrites `readUrl` to implement caching. 
With this change this low level function can implement caching and reuse the logic in the higher level functions like `readUrl` or `readTree`

Reply:
I'd prefer to not do that. We pretty strongly favor composition over inheritance, and this is not a significant savings it seems to me. @backstage/maintainers

Reply:
In that case, what are your thougths in creating a http interface in `backend-common`, which will be used instead of fetch. 

This interface can be than overwritten by devs or customized to implement low level caching, observability and such.

Reply:
We do have `fetchApiRef` in the frontend, maybe this does make some sense in the backend eventually, but it's a gonna be pretty hard I think to motivate plugin authors and prove the value of it if it's just a wrapper around `fetch`.

@freben what's your thoughts here? Don't have a strong opinion either way.

Reply:
I have removed the API change, so this should unblock this PR. 

Currently we are using our own custom GitHubUrlReader, but we would prefer to use an upstream solution and to invest the engineering time rather there.